### PR TITLE
resolve currentPath

### LINKED ISSUE #4496: Build output is different when building on Windows or Mac
### Rollup Version

2.70.1

### Operating System (or Browser)

Windows

### Node Version (if applicable)

14.16.1

### Link To Reproduction

not available

### Expected Behaviour

Seems like on Windows, **preserveModulesRoot: 'src'** will only work on entry files (files defined in "input"). While on OSX it will work on all files.

I would expect it to be consistent between different environments.

Found this question but there is no working answer:
https://stackoverflow.com/questions/64570174/rollup-preservemodulesroot-not-working-as-expected 

### Actual Behaviour

I have a build output defined like:

```
{
  input: 'src/index.ts',
  output: [
    {
      dir: 'dist/cjs',
      format: 'cjs',
      preserveModules: true,
      preserveModulesRoot: 'src',
      sourcemap: true,
      exports: 'named',
    },
    {
      dir: 'dist/esm',
      format: 'esm',
      preserveModules: true,
      sourcemap: true,
      preserveModulesRoot: 'src',
      exports: 'named',
    },
  ],
}
```

When building on Windows, the output in **dist** is like:

```
dist
    -- esm
        --components
        --src
            --components
       index.d.ts
       index.js

```
esm/components will only include d.ts files and esm/src/components will actually include .js files

When building on OSX, the output in **dist** is like:

```
dist
    -- esm
        --components
       index.d.ts
       index.js
```

notice there is no "src" folder and all files are placed inside esm/components.

### PR DESCRIPTION
**Pull request by cleverpp**
This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

- resolves #4496

### CODE REVIEW

**Comment by lukastaegert** [comment]
Can you add a test that is fixed by this change?

**Comment by cleverpp** [comment]
> Can you add a test that is fixed by this change?

you mean in test directory add test case？ 

i had found test case “test/chunking-form/samples/preserve-modules-root “， it do pass in windows。

should i need add a new test case？

**Comment by lukastaegert** [comment]
Yes. The old test case was green without your change, so it was not good enough. We need a test that does not pass without your change. It would also help me and others understand the issue better, is useful for debugging and of course prevents us from breaking it again if we change the logic later

**Comment by Tanimodori** [comment]
I've digged more deeply into this issue that it turns out to be an issue with `vite`'s id normalization.

**TL;DR Vite normalize all file ids to use slash `/` on all platforms, but `rollup` uses `path.resolve` which is platform-dependent (backslash `\\` for Windows). Which causes this problem.**

**Comment by Tanimodori** [comment]
Related issues: https://github.com/rollup/rollup/issues/4496 https://github.com/vitejs/vite/issues/6509

**Comment by lukastaegert** [comment]
I added the test myself...
